third_party/kubetest2/pkg/tester/e2e-tester.go

```diff
@@ -0,0 +1,142 @@
+	Kubeconfig              string `desc:"Path to a cluster kubeconfig file (optional)"`
+	Feature                 string `desc:"Regular expression to select feature(s) to test"`
+	Assess                  string `desc:"Regular expression to select assessment(s) to run"`
+	Labels                  string `desc:"Comma-separated key=value to filter features by labels"`
+	SkipLabels              string `desc:"Regular expression to skip label(s) to run"`
+	SkipFeatures            string `desc:"Regular expression to skip feature(s) to run"`
+	SkipAssessment          string `desc:"Regular expression to skip assessment(s) to run"`
+	Parallel                bool   `desc:"Run test features in parallel"`
+	DryRun                  bool   `desc:"Run Test suite in dry-run mode. This will list the tests to be executed without actually running them"`
+	FailFast                bool   `desc:"Fail immediately and stop running untested code"`
+	DisableGracefulTeardown bool   `desc:"Ignore panic recovery while running tests. This will prevent test finish steps from getting executed on panic"`
+	Packages                string `desc:"Space-separated packages to test"`
+}
+
+const usage = `Usage: kubetest2-tester-e2e-framework [e2e-framework-flags]
+When used with kubetest2: kubetest2 [<deployer>] --test=e2e-framework -- [e2e-framework-flags]
+e2e-framework flags:
+`
+
+func (t *Tester) Execute(args []string) error {
+	fs, err := gpflag.Parse(t)
+	if err != nil {
+		return fmt.Errorf("failed to initialize e2e-framework tester: %s", err)
+	}
+
+	fs.AddGoFlagSet(flag.CommandLine)
+
+	help := fs.BoolP("help", "h", false, "")
+	if err := fs.Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %s", err)
+	}
+
+	fs.Usage = func() {
+		fmt.Print(usage)
+	}
+
+	if *help {
+		fs.SetOutput(os.Stdout)
+		fs.Usage()
+		fs.PrintDefaults()
+		return nil
+	}
+
+	if err := testers.WriteVersionToMetadata(GitTag); err != nil {
+		return err
+	}
+	return t.Test()
+}
+
+func (t *Tester) Test() error {
+	testCmd := t.buildCmd()
+	fmt.Println("Running: ", testCmd)
+	p := gexe.NewProc(testCmd)
+	p.SetStdout(os.Stdout)
+	p.SetStderr(os.Stderr)
+	return p.Run().Err()
+}
+
+func (t *Tester) buildCmd() string {
+	var testCmd strings.Builder
+	testCmd.WriteString(fmt.Sprintf("go test %s %s -args", t.TestFlags, t.Packages))
+	if t.Namespace != "" {
+		testCmd.WriteString(" --namespace=" + t.Namespace)
+	}
+	if t.Kubeconfig != "" {
+		testCmd.WriteString(" --kubeconfig=" + t.Kubeconfig)
+	}
+	if t.Feature != "" {
+		testCmd.WriteString(" --kubeconfig=" + t.Feature)
+	}
+	if t.SkipFeatures != "" {
+		testCmd.WriteString(" --skip-features=" + t.SkipFeatures)
+	}
+	if t.Assess != "" {
+		testCmd.WriteString(" --assess=" + t.Assess)
+	}
+	if t.SkipAssessment != "" {
+		testCmd.WriteString(" --skip-assessment=" + t.SkipAssessment)
+	}
+	if t.Labels != "" {
+		testCmd.WriteString(" --labels=" + t.Labels)
+	}
```

Comment:
Is there a way to simplify this by keeping this set of things co-located in the flags itself ? Will make it easy when someone is adding a new Argument for the `e2e-framework`.

Reply:
I have thought about the exact same thing, basically creating a new type (like a map) that has all the info and be used in both places (here and in the `pkg/flags` packages). One problem is that the `flag` package works differently then the `gpflag` package. So some amount of repeat will be needed.  

For now, I'd like to keep this implementation simple without spreading more complexity in the PR. This can be revisited in a new PR later.

Reply:
/Ack. We can do that for sure.